### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by bizip**
**What does this PR do?**

- Users

A method that returns the 3 most recent posts for a given user.

- Posts

A method that updates the posts counter for a user.
A method that returns the 5 most recent comments for a given post.

- Comments

A method that updates the comments counter for a post.

- Likes

A method that updates the likes counter for a post.

- All columns that are foreign keys should have a corresponding index.

**Description of Task to be completed**

All tasks are completed 

**How should this be manually tested?**

1. Clone this repository and switch to `process-data`

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ActiveRecord::Base
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (rloterh) [inline app/models/user.rb]: - [ ] As per the given _ERD_, the _User_ **association** should have _author_id_ referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve this aspect, similar to the code example below:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Post < ActiveRecord::Base
+  belongs_to :user
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] As per the given _ERD_, the _Post_ **association** should have _Post_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Comment < ActiveRecord::Base
+  belongs_to :user
```
Comment (rloterh) [inline app/models/comment.rb]: - [ ] As per the given _ERD_, the _Comment_ **association** should have _Comment_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Like < ActiveRecord::Base
+  belongs_to :user
```
Comment (rloterh) [inline app/models/like.rb]: - [ ] As per the given _ERD_, the _Like_ **association** should have _Like_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  has_many :comments
+  has_many :likes
+  def update_post_counter
+    author.increment!(:postsCountter)
+  end
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_posts_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private, similar to:

```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+  def update_comment_counter
+    posts.increment!(:commentsCounter)
+  end
```
Comment (rloterh) [inline app/models/comment.rb]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_comments_counter method every time a new like has been saved to database, then make the `update_comments_counter` method private, similar to:

```ruby
 # app/models/comments.rb
  after_save :update_comments_counter

  private

  def update_comment_counter
    author.increment!(:comments_counter)
  end
```

File: app/models/like.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+  belongs_to :Post
+
+  def update_like_counter
+    posts.increment!(:likesCounter)
+  end
```
Comment (rloterh) [inline app/models/like.rb]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_likes_counter method every time a new like has been saved to database, then make the `update_likes_counter` method private, similar to:

```ruby
 # app/models/likes.rb
  after_save :update_like_counter

  private

  def update_like_counter
    author.increment!(:likes_counter)
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Pascal, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Descriptive commit messages


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
Hi @bizip 🙋‍♂️,

Code looks good 👍

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**